java/sqlcommenter-java/src/main/java/com/google/cloud/sqlcommenter/threadlocalstorage/SpanContextMetadata.java

```diff
@@ -0,0 +1,83 @@
+package com.google.cloud.sqlcommenter.threadlocalstorage;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class SpanContextMetadata {
+    private static final Logger logger = Logger.getLogger(SpanContextMetadata.class.getName());
+
+    private boolean isValid = true;
+    private String traceId;
+    private String spanId;
+    private byte traceOptions;
+    private String traceState;
+
+    public SpanContextMetadata(io.opencensus.trace.SpanContext spanContext) {
+        if (spanContext == null || !spanContext.isValid()) {
+            this.isValid = false;
+            return;
+        }
+        this.traceId = spanContext.getTraceId().toLowerBase16();
+        this.spanId = spanContext.getSpanId().toLowerBase16();
+        this.traceOptions = spanContext.getTraceOptions().getByte();
+
+        io.opencensus.trace.Tracestate traceState = spanContext.getTracestate();
+        if (traceState.getEntries().isEmpty()) {
+            this.traceState = null;
+            return;
+        }
+
+        // Tracestate needs to be serialized in the order of the entries.
```

Comment:
It's a bad practice for constructors to have logic, so I'd suggest to rewrite the code in this class to do the following:

```java
private SpanContextMetadata(boolean isValid, String traceId, String spanId, byte traceOptions, String traceState) {
  this.isValid = isValid;
  this.traceId = traceId;
  this.spanId = spanId;
  this.traceOptions = traceOptions;
  this.traceState = traceState;
}

public static SpanContextMetadata fromOpenCensusContext(io.opencensus.trace.SpanContext spanContext) {
  ...
  return new SpanContextMetadata(...);
}


public static SpanContextMetadata fromOpenTelemetryContext(io.opentelemetry.api.trace.SpanContext spanContext) {
  ...
  return new SpanContextMetadata(...);
}
```